GCTC-NTGC/gc-digital-talent
# Update user email, full name fallbacks

Resolves #3961. h/t @esizer.

## Summary
The **Admin > Users table** _Email_ and _Candidate Name_ values now have human-readable null state values. Additionally, the following are now more human-readable and consistent since they use the same function that uses the same logic for getting the "full name":

- **Admin > Dashboard** heading
- **Admin > Candidate Details** heading and breadcrumbs
- **Admin > Candidate Details > General Information > About section** name value (note: the logic surrounding this key and value is out of scope of this PR: https://github.com/GCTC-NTGC/gc-digital-talent/issues/4075)
- **Admin > Candidate Details > Candidate Profile > About Me section** name value (note: logic surrounding this key and value is out of scope of this PR: https://github.com/GCTC-NTGC/gc-digital-talent/issues/4075)
- **Admin > Candidate Details > Candidate Status section > Change status dialog** name value
- **Profile** heading
- **Profile > About section** name value
- **Search > Search Pools section** Pool Owner value

Also, the following was added as part of the PR:
- [Adds utility functions for getting full names](https://github.com/GCTC-NTGC/gc-digital-talent/commit/2c4f76b5d806366f505db77bbf6a7572048ec493)
- [Replaces logic for getting full name with getFullNameHtml, getFullNameLabel](https://github.com/GCTC-NTGC/gc-digital-talent/commit/64792ac9fc2917b9b83437742b7402f657513283) 
- [Removes display, align-items css attributes for non-link breadcrumbs to avoid elements sitting directly beside each other without space for when those elements contain a child span](https://github.com/GCTC-NTGC/gc-digital-talent/commit/235bfa1e1190e9ad3ef5f6a60a4c51a5f3d24e9d)

## Cases with expected copy
- No email: _No email provided_
- No first name or last name: _No name provided_
- No first name: _No first name provided_ Lastname
- No last name: Firstname _No last name provided_

## Testing
1. Build admin `npm run production --workspace=admin`
2. Build talent search `npm run production --workspace=talentsearch`
3. Navigate to and login at /admin
4. Browse users table in database and find user that you are logged in as (preferably, the admin@test.com user)
5. Set each of first_name, last_name, and email fields to `NULL`
6. Navigate to Dashboard (admin)
7. Verify user that you are logged in as has expected copy (see cases above) as the value in the Welcome heading
8. Navigate to Users table (admin)
9. Verify user that you are logged in as has expected copy (see cases above) as the value under Email column
10. Verify user that you are logged in as has expected copy (see cases above) as the value under Candidate Name column
11. Navigate to Candidate Details by clicking on user table email column value (admin)
12. Verify user that you are logged in as has expected copy (see cases above) as the value in the last breadcrumb and the heading
13. Navigate to My Profile (talent search)
14. Verify Profile heading has expected copy (see cases above)
15. Verify Profile About section has expected copy (see cases above)
16. Navigate to Search (talent search)
17. Verify Pool Owner: value has expected copy (see cases above)
18. Repeat steps 6-17 while changing first_name to a value other than `NULL` and last_name as `NULL`
19. Repeat steps 6-17 while changing first_name to `NULL` and last_name to a value other than `NULL`
